flink-runtime/src/main/java/org/apache/flink/runtime/operators/BatchTask.java

```diff
@@ -1250,8 +1250,11 @@ public static void logAndThrowException(Exception ex, AbstractInvokable parent)
 				oe = new OutputEmitter<T>(strategy, indexInSubtaskGroup, comparator, partitioner, dataDist);
 			}
 
-			final RecordWriter<SerializationDelegate<T>> recordWriter =
-					new RecordWriter<SerializationDelegate<T>>(task.getEnvironment().getWriter(outputOffset + i), oe);
+			final RecordWriter<SerializationDelegate<T>> recordWriter = RecordWriter.createRecordWriter(
+				task.getEnvironment().getWriter(outputOffset + i),
+				oe,
+				-1,
```

Comment:
another nit/tiny issue: here `-1` constant is leaking from the `RecordWriter`. Maybe overloading `createRecordWriter()` with default value of the `timeout`?